tgui/packages/tgui/interfaces/PaiInterface.tsx

```diff
@@ -0,0 +1,540 @@
+  const { act, data } = useBackend<PaiInterfaceData>(context);
+  const { languages, pda } = data;
+  const { software } = props;
+
+  switch (software) {
+    case 'digital messenger':
+      return (
+        <>
+          <Button
+            icon="power-off"
+            onClick={() => act('pda', { pda: 'power' })}
+            selected={!pda.power}>
+            Power
+          </Button>
+          <Button
+            icon="volume-mute"
+            onClick={() => act('pda', { pda: 'silent' })}
+            selected={pda.silent}>
+            Silent
+          </Button>
+          <Button
+            icon="envelope"
+            onClick={() => act('pda', { pda: 'message' })}>
+            Message
+          </Button>
+        </>
+      );
+    case 'door_jack':
+      return (
+        <>
+          <Button
+            icon="power-off"
+            onClick={() => act('door_jack', { jack: 'cable' })}
+            selected={!pda.power}>
+            Hack
+          </Button>
+          <Button
+            icon="power-off"
+            onClick={() => act('door_jack', { jack: 'jack' })}
+            selected={!pda.power}>
+            Hack
+          </Button>
+          <Button
+            icon="power-off"
+            onClick={() => act('door_jack', { jack: 'cancel' })}
+            selected={!pda.power}>
+            Hack
+          </Button>
+        </>
+      );
+    case 'universal translator':
+      return (
+        <Button
+          icon="download"
+          onClick={() => act(software.toLowerCase().replace(/ /g, '_'))}
+          disabled={!!languages}>
+          {!languages ? 'Install' : 'Installed'}
+        </Button>
+      );
+    default:
+      return (
+        <Button
+          icon="power-off"
+          onClick={() => act(software.toLowerCase().replace(/ /g, '_'))}
+          tooltip="Attempts to toggle the module's power.">
+          Toggle
+        </Button>
+      );
+  }
+};
+
+const AvailableDisplay = () => {
+  return (
+    <Section
+      buttons={<AvailableMemory />}
+      fill
+      scrollable
+      title="Available Software">
+      <AvailableSoftware />
+    </Section>
+  );
+};
+
+const AvailableMemory = (_, context) => {
+  const { data } = useBackend<PaiInterfaceData>(context);
+  const { ram } = data;
+
+  return (
+    <Tooltip content="Available System Memory">
+      <Stack>
+        <Icon color="purple" mt={0.7} mr={1} name="microchip" />
+        <ProgressBar
+          minValue={0}
+          maxValue={100}
+          ranges={{
+            good: [67, 100],
+            average: [34, 66],
+            bad: [0, 33],
+          }}
+          value={ram}
+        />
+      </Stack>
+    </Tooltip>
+  );
+};
+
+const AvailableSoftware = (_, context) => {
+  const { act, data } = useBackend<PaiInterfaceData>(context);
+  const { available, installed } = data.software;
+  /** Converted to type: Available */
+  let convertedList: Available[] = Object.keys(available).map((key) => {
```

Comment:
Make this `const`, and just use `Object.entries` instead, which gives you both key and value.

Reply:
From there you could likely remove `Available` altogether, or at least change to `type Available = (K, V);`